### PR TITLE
[Kraken] Add address in stop point

### PR DESCRIPTION
**Pull request by azime**

### CODE REVIEW

**Review by pbench — APPROVED** [review]

File: source/type/data.cpp [hunk 1]
```diff
@@ -484,6 +489,26 @@ void Data::aggregate_odt() {
     }
 }
 
+void Data::fill_stop_point_address() {
+    size_t without_address = 0;
+    for (auto sp : pt_data->stop_points) {
```
Comment (benoit-bst) [inline source/type/data.cpp:494]:
```suggestion
    for (const auto& sp : pt_data->stop_points) {
```
?

File: source/type/data.cpp [hunk 2]
```diff
@@ -484,6 +489,26 @@ void Data::aggregate_odt() {
     }
 }
 
+void Data::fill_stop_point_address() {
+    size_t without_address = 0;
```
Comment (benoit-bst) [inline source/type/data.cpp:493]: _unsigned_ is better ? like **uint32** to avoid overflow?

File: source/type/data.cpp (hunk 1 above)
Comment (azime) [reply source/type/data.cpp:494]: No I modify it, [see](https://github.com/CanalTP/navitia/pull/3434/files#diff-b18f3fcd9eef06ffc9132fca1d60bdfcf7c92bf0bd36f2fb29a554b679e2d2f8R501)

**Review by kadhikari — APPROVED** [review]
Great :+1:

**Review by benoit-bst — APPROVED** [review]

File: source/type/data.cpp [hunk 3]
```diff
@@ -484,6 +489,26 @@ void Data::aggregate_odt() {
...
+        try {
+            auto nb_way = geo_ref->nearest_addr(sp->coord);
+            sp->address = new navitia::georef::Address(nb_way.second, sp->coord, nb_way.first);
+        } catch (const navitia::proximitylist::NotFound&) {
+            LOG4CPLUS_DEBUG(log4cplus::Logger::getInstance("logger"),
+                            "unable to find a way from coord [" << sp->coord.lon() << "-" << sp->coord.lat() << "]");
+            ++without_address;
+        }
+    }
+    if (without_address) {
```
Comment (xlqian) [inline source/type/data.cpp:508]: in this case, please use boost::optional

File: source/type/data.cpp (hunk 2 above)
Comment (AzHicham) [reply source/type/data.cpp:493]: std::size_t is already unsigned I think. (could be u32 or u64 depending on the platform)

**Pull request merged**